lib/nat/service.go

```diff
@@ -249,10 +250,20 @@ func (s *Service) verifyExistingLocked(ctx context.Context, mapping *Mapping, na
 
 			l.Debugf("Renewing %s -> %v open port on %s", mapping, extAddrs, id)
 
+			if mapping.ipVersion == IPv4Only && nat.IsIPv6GatewayDevice() {
+				l.Debugf("Skipping renew for %s because listener is IPv4-only", nat.ID())
+				continue
+			}
+
+			if mapping.ipVersion == IPv6Only && !nat.IsIPv6GatewayDevice() {
```

Comment:
Wouldn't it be easier if the interface had a `SupportsIPVersion` method instead, re-using the new `IPVersion` type?

And the whole "Skipping ..." block should probably go before the debug message stating that it is being renewed.

Reply:
Done, although I find that having to implement the logic for every type of NAT service is probably not so nice either and due to the lack of default implementations in the language it can't be worked around by writing one that checks for `IsIPv6GatewayDevice`. This check in`lib/nat/service.go` also doesn't look clean to me, so I'd keep this tradeoff (and maybe replace `IsIPv6GatewayDevice` with `SupportsIPVersion`, although I'd find it awkward to have to call it as `nat.SupportsIPVersion(nat.IPv6Only)` in `tryNATDevice` vs. the call without the parameter)

Reply:
Well I didn't mean to duplicate `IsIPv6GatewayDevice` with another wrapper, but rather completely replace it.  I admit my suggested naming is not good after looking at all the calling sites.  Sorry my time slots for review are currently short until getting too tired to grasp the whole picture anymore.  Will try to revise the suggestion in the hope to make it easier to understand what the checks imply.